internal/db/schema/migrations/postgres/11/01_server_type_enum_test.go

```diff
@@ -0,0 +1,191 @@
+	rootWrapper := db.TestWrapper(t)
+	repo := servers.TestRepo(t, conn, rootWrapper)
+
+	// Seed the data
+	origController := &servers.Server{
+		PrivateId: "test-controller",
+		Type:      "controller",
+		Address:   "127.0.0.1",
+	}
+	_, _, err = repo.UpsertServer(ctx, origController)
+	require.NoError(err)
+
+	origWorker := &servers.Server{
+		PrivateId: "test-worker",
+		Type:      "worker",
+		Address:   "127.0.0.1",
+		Tags: map[string]*servers.TagValues{
+			"tag": {
+				Values: []string{"value1", "value2"},
+			},
+		},
+	}
+	_, _, err = repo.UpsertServer(ctx, origWorker, servers.WithUpdateTags(true))
+	require.NoError(err)
+
+	// Read the servers back
+	expectedControllers, err := repo.ListServers(ctx, servers.ServerTypeController)
+	require.NoError(err)
+	require.Len(expectedControllers, 1)
+
+	expectedWorkers, err := repo.ListServers(ctx, servers.ServerTypeWorker)
+	require.NoError(err)
+	require.Len(expectedWorkers, 1)
+
+	// now we're ready for the migration we want to test.
+	oState = schema.TestCloneMigrationStates(t)
+	nState = schema.TestCreatePartialMigrationState(oState["postgres"], serverEnumMigration)
+	oState["postgres"] = nState
+
+	m, err = schema.NewManager(ctx, dialect, d, schema.WithMigrationStates(oState))
+	require.NoError(err)
+
+	require.NoError(m.RollForward(ctx))
+	state, err = m.CurrentState(ctx)
+	require.NoError(err)
+	require.Equal(serverEnumMigration, state.DatabaseSchemaVersion)
+	require.False(state.Dirty)
+
+	t.Log("current migration version: ", state.DatabaseSchemaVersion)
+
+	// Assert servers
+	actualControllers, err := repo.ListServers(ctx, servers.ServerTypeController)
+	require.NoError(err)
+	require.Len(actualControllers, 1)
+	require.Equal(expectedControllers, actualControllers)
+
+	actualWorkers, err := repo.ListServers(ctx, servers.ServerTypeWorker)
+	require.NoError(err)
+	require.Len(actualWorkers, 1)
+	require.Equal(expectedWorkers, actualWorkers)
+
+	// Assert the state of the newly created enum table
+	rows, err := d.QueryContext(ctx, "select * from server_type_enm")
+	require.NoError(err)
+
+	// Column types
+	colTypes, err := rows.ColumnTypes()
+	require.NoError(err)
+	require.Len(colTypes, 1)
+	require.Equal("TEXT", colTypes[0].DatabaseTypeName())
+	require.Equal("name", colTypes[0].Name())
+	if nullable, ok := colTypes[0].Nullable(); ok {
+		require.Equal(false, nullable)
+	}
+
+	// Inserted rows
+	var actualEnm []string
+	for rows.Next() {
+		var a string
+		require.NoError(rows.Scan(&a))
+		actualEnm = append(actualEnm, a)
+	}
+	require.Equal([]string{"controller", "worker"}, actualEnm)
+
+	// Try inserting a broken row
+	result, err := d.ExecContext(ctx, "insert into server_type_enm values ($1)", "bad")
+	require.EqualError(err, expectEnumConstraintErr)
+	require.Nil(result)
+
+	// Try adding a broken server type
+	badServer := &servers.Server{
+		PrivateId: "test-bad",
+		Type:      "bad",
+		Address:   "127.0.0.1",
+	}
+	_, rowsUpdated, err := repo.UpsertServer(ctx, badServer)
```

Comment:
Perhaps consider not using the repo.UpsertServer here and creating the dep between this test and how the repo function is implemented.  This sort of applies to all these repo functions...